### PR TITLE
Update table framework

### LINKED ISSUE #223: Search sample table doesnt recognise styled cells
The search functionality of `w2grid` have issues with styled cells. In addition doesnt it seem to be very well maintained and the documentation is a bit limited.

We better switch to another table library.

### PR DESCRIPTION
**Pull request by mhkc**
This PR replaces w2ui with data tables.

closes #223

### CODE REVIEW

File: frontend/bonsai_app/blueprints/groups/static/datatables.min.css
Comment (ryanjameskennedy) [inline frontend/bonsai_app/blueprints/groups/static/datatables.min.css:1]: Have you copied and pasted this?

File: frontend/bonsai_app/blueprints/groups/static/jquery-3.7.1.min.js
Comment (ryanjameskennedy) [inline frontend/bonsai_app/blueprints/groups/static/jquery-3.7.1.min.js:1]: Is this the same as installing jquery?

File: frontend/bonsai_app/blueprints/groups/templates/group.html [hunk 1]
```diff
@@ -1,13 +1,18 @@
 {% extends "layout.html" %}
-{% import "cell_fmt.html" as cell_format %}
-{% from "shared.html" import search_similar_btn, search_similar_js, add_to_basket_btn, add_to_basket_js, qc_bulk_toggle, delete_samples_btn %}
+{% from "shared.html" import search_similar_btn, search_similar_js, add_to_basket_btn, add_to_basket_js, qc_bulk_toggle, delete_samples_btn, sample_table %}
 
 {% block css %}
-    {{ super() }}
-    <link rel="stylesheet" type="text/css" href="https://rawgit.com/vitmalina/w2ui/master/dist/w2ui.min.css">
-    <link rel="stylesheet" type="text/css" href="{{ url_for('groups.static', filename='group.css') }}">
+{{ super() }}
```
Comment (ryanjameskennedy) [inline frontend/bonsai_app/blueprints/groups/templates/group.html:5]: No indent any more?

File: frontend/bonsai_app/blueprints/groups/templates/groups.html [hunk 2]
```diff
@@ -1,12 +1,18 @@
 {% extends "layout.html" %}
-{% from "shared.html" import search_similar_btn, search_similar_js, add_to_basket_btn, add_to_basket_js, qc_bulk_toggle, delete_samples_btn %}
+{% from "shared.html" import search_similar_btn, search_similar_js, add_to_basket_btn, add_to_basket_js, qc_bulk_toggle, delete_samples_btn, sample_table %}
 
 {% block css %}
 {{ super() }}
-<link rel="stylesheet" type="text/css" href="https://rawgit.com/vitmalina/w2ui/master/dist/w2ui.min.css">
```
Comment (ryanjameskennedy) [inline frontend/bonsai_app/blueprints/groups/templates/groups.html:6]: Can't we do this instead of jquery.min.css file

**Review by ryanjameskennedy — APPROVED** [review]
LGTM!

File: frontend/bonsai_app/blueprints/groups/static/datatables.min.css
Comment (mhkc) [reply frontend/bonsai_app/blueprints/groups/static/datatables.min.css:1]: No its just a regular minified css from the datatables assets builder

File: frontend/bonsai_app/blueprints/groups/static/jquery-3.7.1.min.js
Comment (mhkc) [reply frontend/bonsai_app/blueprints/groups/static/jquery-3.7.1.min.js:1]: yes. Its a local install of the version of jquery used by datatables.

**Pull request merged**